File: executor/src/main/java/com/codeloom/executor/engine/SubmissionContext.java
package com.codeloom.executor.engine;

import com.codeloom.common.language.LanguageSpec;
import java.util.UUID;
import lombok.Builder;

@Builder
public record SubmissionContext(
        UUID submissionId,
        UUID userId,
        long problemId,
        String code,
        LanguageSpec language,
        Long executionTimeLimitMs,
        Long memoryUsageLimitBytes) {}

File: executor/src/main/kotlin/com/codeloom/executor/engine/SubmissionContext.kt
package com.codeloom.executor.engine

import com.codeloom.common.language.LanguageSpec
import java.util.*

data class SubmissionContext(
    val submissionId: UUID,
    val userId: UUID,
    val problemId: Long,
    val code: String,
    val language: LanguageSpec,
    val executionTimeLimitMs: Long? = null,
    val memoryUsageLimitBytes: Long? = null,
)
